core/camel-support/src/main/java/org/apache/camel/support/RestConsumerContextPathMatcher.java

```diff
@@ -331,11 +345,31 @@ private static int countWildcards(String consumerPath) {
         String[] consumerPaths = consumerPath.split("/");
         for (String p2 : consumerPaths) {
             if (p2.startsWith("{") && p2.endsWith("}")) {
-                wildcards++;
+                curlyBraces++;
             }
         }
 
-        return wildcards;
+        return curlyBraces;
+    }
+
+    private static boolean matchWildCard(String requestPath, String consumerPath) {
+        if (!requestPath.endsWith("/")) {
+            requestPath = requestPath + "/";
+        }
+        // Convert URI template to a regex pattern
+        String regex = consumerPath
+                .replace("/", "\\/")
+                .replace("{", "(?<")
+                .replace("}", ">[^\\/]+)");
+
+        // Add support for wildcard * as path suffix
+        regex = regex.replace("*", ".*");
+
+        // Match the provided path against the regex pattern
```

Comment:
Is there a way to avoid regexp compile, as its slow and would impact high performance.
We need to do a simpler way of checking this.

Reply:
> Is there a way to avoid regexp compile, as its slow and would impact high performance. We need to do a simpler way of checking this.

Should be precompiled and saved, rather than compiling every request.

Reply:
> Is there a way to avoid regexp compile, as its slow and would impact high performance. We need to do a simpler way of checking this.

Trie tree is suitable for route matching, but it may require substantial code modifications.

Reply:
Ah okay pre-compiled really helps, so its only calling the matcher per message